src/libraries/System.Linq.Expressions/src/ILLink/ILLink.Substitutions.xml

```diff
@@ -0,0 +1,13 @@
+<linker>
+  <assembly fullname="System.Linq.Expressions">
+    <type fullname="System.Linq.Expressions.LambdaExpression">
+      <method signature="System.Boolean get_CanCompileToIL()" feature="System.Linq.Expressions.CanCompileToIL" featurevalue="false" body="stub" value="false" />
```

Comment:
https://github.com/dotnet/runtime/issues/17973#issuecomment-277157066 has reasoning for why this needs to be different from RuntimeFeature.IsDynamicCodeSupported.

> Also the defaults don't look correct for existing configurations (e.g. published trimmed console app).

Could you clarify?

Reply:
>  has reasoning for why this needs to be different from

I don't see the reasoning in the comment and this mode is not new in .NET7. In .NET6 we use the same options to detect NativeAOT like setup where IsDynamicCodeSupported indicates that code generation is possible with IsDynamicCodeCompiled specifying how it such generated code processed. 

>  Could you clarify?

What does happen today when you publish trimmed console for desktops. I guess it has "CanCompileToIL" set to true and here we are setting it to false for everything.

Reply:
> I don't see the reasoning in the comment and this mode is not new in .NET7. In .NET6 we use the same options to detect NativeAOT like setup where IsDynamicCodeSupported indicates that code generation is possible with IsDynamicCodeCompiled specifying how it such generated code processed.

We could make this so that the body of `CanCompileToIL` is implemented as `RuntimeFeature.IsDynamicCodeSupported`, but I'm a believer in separating behavioral changes from refactoring changes and this would be a behavioral change. It can happen in subsequent pull request as a thing that can be discussed separately from this refactoring.

> What does happen today when you publish trimmed console for desktops. I guess it has "CanCompileToIL" set to true and here we are setting it to false for everything.

The substitution is set as `feature="X" featurevalue="false" body="stub" value="false"` - if I'm reading IL Linker sources right, this substitution will be ignored unless someone sets feature X to false. Is that correct reading? I wasn't able to build intuition behind what `featurevalue=X featuredefault=Y` means and always defer to reading linker source code. If my interpretation is correct, this will do the right thing for trimmed apps (IL compiler will be kept).